web/src/js/backends/websocket.tsx

```diff
@@ -94,6 +94,27 @@ export default class WebsocketBackend {
         }
     }
 
+    updateFilter(name: string, expr: string) {
+        this.sendMessage("flows", {
+            cmd: "updateFilter",
+            name,
+            expr,
+        });
+    }
+
+    private sendMessage(resource, data) {
+        if (this.socket && this.socket.readyState === WebSocket.OPEN) {
```

Comment:
What about `WebSocket.CONNECTING`? We probably should collect messages during connection establishment and then send them in onOpen. The filter state from the URL will come very early, likely before a connection has been established.

Reply:
can't we do something like this?
`if (
            this.socket &&
            (this.socket.readyState === WebSocket.CONNECTING ||
                this.socket.readyState === WebSocket.OPEN)
        ) {...}`

Reply:
Can you already send on a socket while it is still in CONNECTING state? If that works that's great, but I don't know. :)

Reply:
If I do something like this: 
`if (this.socket && this.socket.readyState === WebSocket.OPEN) {...}` and then refresh or change the url, I get an error saying that I cannot send messages because the socket is not open; while if I do something like this:
`if ( this.socket && (this.socket.readyState === WebSocket.CONNECTING || this.socket.readyState === WebSocket.OPEN) ) {...}`, it works fine 😅